app/models/product.rb

```diff
@@ -0,0 +1,32 @@
+class Product < ActiveRecord::Base
+  belongs_to :user
+  has_many :order_items
+  has_many :reviews
+  has_and_belongs_to_many :categories
+  #before_validation :dollars_to_cents
+
+  validates :name, length: { :maximum => 30 }
```

Comment:
Do you expect all product names to be fewer than 30 characters long? If not, you're restricting users' freedom to name things and not gaining much. Reasons to restrict characters would be to conserve space in a DB, or if you know a particular field will only ever have a certain size (eg. logins always being 8 or fewer characters).